src/main/java/com/ita/if103java/ims/dao/impl/AddressDaoImpl.java

```diff
@@ -0,0 +1,216 @@
+                               String crudErrorMessage) {
+        try {
+            final KeyHolder keyHolder = new GeneratedKeyHolder();
+            namedJdbcTemplate.update(query, sqlParameterSource, keyHolder);
+            return Optional
+                .ofNullable(keyHolder.getKey())
+                .map(Number::longValue)
+                .orElseThrow(() -> new CRUDException(crudErrorMessage));
+        } catch (DataAccessException e) {
+            throw new CRUDException(crudErrorMessage, e);
+        }
+    }
+
+    public void updateSingleAddressRecord(String query, MapSqlParameterSource sqlParameterSource,
+                                          String notFoundMessage, String crudErrorMessage) {
+        try {
+            final int rowsAffected = namedJdbcTemplate.update(query, sqlParameterSource);
+            if (rowsAffected == 0) {
+                throw new EntityNotFoundException(notFoundMessage);
+            }
+        } catch (DataAccessException e) {
+            throw new CRUDException(crudErrorMessage, e);
+        }
+    }
+
+    public Address findSingleAddressRecord(String query, MapSqlParameterSource sqlParameterSource,
+                                           String notFoundMessage, String crudErrorMessage) {
+        try {
+            return namedJdbcTemplate.queryForObject(query, sqlParameterSource, mapper);
+        } catch (EmptyResultDataAccessException e) {
+            throw new EntityNotFoundException(notFoundMessage);
+        } catch (DataAccessException e) {
+            throw new CRUDException(crudErrorMessage, e);
+        }
+    }
+
+    private MapSqlParameterSource getSqlParameterSource(Address address) {
+        return new MapSqlParameterSource()
+            .addValue("country", address.getCountry())
+            .addValue("city", address.getCity())
+            .addValue("address", address.getAddress())
+            .addValue("zip", address.getZip())
+            .addValue("latitude", address.getLatitude())
+            .addValue("longitude", address.getLongitude());
+    }
+
+    public static class Queries {
+        public static final String SQL_CREATE_NEW_WAREHOUSE_ADDRESS = """
+                insert into addresses(warehouse_id, country, city, address, zip, latitude, longitude)
+                values (:warehouse_id, :country, :city, :address, :zip, :latitude, :longitude)
+            """;
+
+        public static final String SQL_CREATE_NEW_ASSOCIATE_ADDRESS = """
+                insert into addresses(associate_id, country, city, address, zip, latitude, longitude)
+                values (:associate_id, :country, :city, :address, :zip, :latitude, :longitude)
+            """;
+
+        public static final String SQL_UPDATE_WAREHOUSE_ADDRESS = """
+                update addresses
+                set country   = :country,
+                    city      = :city,
+                    address   = :address,
+                    zip       = :zip,
```

Comment:
I think we should follow one code style(only JdbcTemplate or NamedParameterJdbcTemplate ).